### DISCUSSION TITLE
视频超过2倍速问题

### DISCUSSION THREAD

**Discussion by sky-279** [the1812/Bilibili-Evolved#1616] [综合讨论 / General]
在播放带有小p的视频（包括播放所有收藏夹和所有稍后再看的情况）情况下，自动切p会导致倍速卡在2.5无法下降（主要是使用快捷键）（手动切p没有问题）（提高倍速没问题）
目前解决这一问题的办法：
1、用鼠标手动选择除了2倍速之外的其他倍速 
2、使用快捷键恢复1倍速后重新加倍速
3、手动切p
希望能够解答

**Comment by LonelySteve** [comment]
1. 首先请明确一下目前使用的版本，如果是 1.12.0，那么可以尝试重装一下脚本，尤其是离线版，重装之前注意导出保存设置，或者直接上最新的预览版，看是否还存在问题
2. 原因在于我合并代码的时候不小心漏传了一个参数，其实最初在设计上就已经避免了这个 BUG
3. 这个问题应该可以在带分 P 的普通视频页中进行复现，而不仅仅是在集合页

>（如果你对这个 BUG 产生的深层原因感兴趣）B 站自己的播放器内部会维护一个当前倍数值，但是脚本不能侵入修改它，而当用户从原生倍速**按顺序**切换到扩展倍速，实际上内部维护的倍数值仍然会停留在最高的原生倍速（典型情况），如果用户从扩展倍速切换回这个最高的原生倍速，就会触发播放器内部的逻辑，它将出于优化目的，对于相同的倍数值设定是不再进行实际倍数设置以及菜单样式更新，判定依据就是它内部维护的当前倍数值，脚本这块对等维护了一个倍数值，换 P 的时候菜单元素其实发生了改变，因此需要重新实例化新的倍数控制器，并传递之前的维护的倍数值（其实原生播放器已经实现了记忆倍数的功能，然而仅限于刷新页面和跨分 P 的情景），然而这个该传递的值在两次 PR 中都没传递，最新稳定版已经热更新修复了这个问题。

**Comment by gcl-l** [comment]
倍速插件推荐是用  video speed controller 这个插件
我没有遇到你这个问题
